vector/src/main/java/im/vector/app/features/home/room/list/RoomSummaryItemFactory.kt

```diff
@@ -55,12 +58,21 @@ class RoomSummaryItemFactory @Inject constructor(private val displayableEventFor
     fun createSuggestion(spaceChildInfo: SpaceChildInfo,
                          suggestedRoomJoiningStates: Map<String, Async<Unit>>,
                          listener: RoomListListener?): VectorEpoxyModel<*> {
+        val error = (suggestedRoomJoiningStates[spaceChildInfo.childRoomId] as? Fail)?.error
         return SpaceChildInfoItem_()
                 .id("sug_${spaceChildInfo.childRoomId}")
                 .matrixItem(spaceChildInfo.toMatrixItem())
                 .avatarRenderer(avatarRenderer)
                 .topic(spaceChildInfo.topic)
-                .buttonLabel(stringProvider.getString(R.string.join))
+                .errorLabel(
+                        error?.let {
+                            stringProvider.getString(R.string.error_failed_to_join_room, errorFormatter.toHumanReadable(it))
+                        }
```

Comment:
ErrorFormatter.toHumanReadable() can take a null param, so you can write:
```kotlin
errorFormatter.toHumanReadable(error)
    ?.let { stringProvider.getString(R.string.error_failed_to_join_room, it }
```

Reply:
A bit surprised, because when I tested it I saw that it produced a generic message. but the code shows something else

Reply:
You are right, my bad